src/app/api/Mock/MockApi.ts

```diff
@@ -210,6 +212,28 @@ export default class NewClient implements API {
         throw new Error('Method not implemented.');
     }
 
+    async createShiftsFromPreviousWeek(shiftCopyDetails: ShiftCopyInstructions): Promise<Shift[]> {
+        const { startOfWeekToCopy, copySelection, startOfWeekToCreate } = shiftCopyDetails;
+        const shiftsToCopy = sheriffShifts.filter(s => moment(s.startDateTime).isSame(startOfWeekToCopy, 'week'));
+        
+        let copiedShifts: Shift[] = [];
+        shiftsToCopy.forEach(shift => {
```

Comment:
`forEach` is very simple and powerful but can lead to some interesting results (not in this case, but when you mutate the array that your forEaching).  However, this approach is in stateful in nature, since we have a variable (`copiedShifts`) outside the closure that we're pushing items into.  A more stateless and functional approach would be to transform the original collection into a collection of new items like so:

```ts
const shiftCopies = shiftsToCopy.map(s=>{
  const newShift: Shift = {
     ...shift,
     id: this.getId(),
     startDateTime: moment(shift.startDateTime).week(moment(startOfWeekToCreate).week()),
     endDateTime: moment(shift.endDateTime).week(moment(startOfWeekToCreate).week()),
     sheriffId: (copySelection !== 'shiftsOnly') ? shift.sheriffId : undefined
}
return newShift;
);

shiftCopies.forEach(s=>sheriffShifts.push(s))

```

> Note that this can actually just be added onto the end of your filter operation that you applied to sheriffShifts.